src/dragonfly/dragonfly/actions/GradientAction.py

```diff
@@ -114,7 +115,6 @@ def setupSubject(self, drone):
         return position_value_subject
 
     def complete(self):
-        self.log_publisher.publish(
-            "Maximum CO2 of {} found at {}, {}".format(self.max_value.value, self.max_value.latitude,
-                                                       self.max_value.longitude))
+        self.log_publisher.publish(String(data=
+            f"Maximum CO2 of {self.max_value.value} found at {self.max_value.latitude}, {self.max_value.longitude}"))
```

Comment:
This was actually a bug that killed the action once complete.

Reply:
Gotta love bugs that do what we want